src/main/java/com/example/parkingapp/service/ParkingAdministrationService.java

```diff
@@ -0,0 +1,53 @@
+package com.example.parkingapp.service;
+
+import com.example.parkingapp.api.rest.exceptions.VehicleNotParkedException;
+import com.example.parkingapp.api.rest.vm.ParkingStatisticsVM;
+import com.example.parkingapp.api.rest.vm.VehicleStatusVM;
+import com.example.parkingapp.config.ParkingRecordStatus;
+import com.example.parkingapp.domain.ParkingRecord;
+import com.example.parkingapp.repository.ParkingRecordRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Service
+public class ParkingAdministrationService {
+    private final ParkingRecordRepository parkingRecordRepository;
+
+    public ParkingAdministrationService(ParkingRecordRepository parkingRecordRepository) {
+        this.parkingRecordRepository = parkingRecordRepository;
+    }
+
+    /**
+     * Calculates earnings made in given day
+     * @param day day
+     * @return earnings
+     */
+    public ParkingStatisticsVM getEarningsForGivenDay(LocalDate day) {
+        return new ParkingStatisticsVM(parkingRecordRepository.sumAllPaymentsWithinDate(day.atStartOfDay(), day.atStartOfDay().plusDays(1)));
+    }
+
+    /**
+     * Checks if vehicle is parked
+     * @param licensePlateNumber license number of vehicle
+     * @return status
+     */
+    public VehicleStatusVM checkVehicleStatus(String licensePlateNumber) {
+        Optional<ParkingRecord> parkingRecord = parkingRecordRepository.findByLicensePlateAndStatus(licensePlateNumber, ParkingRecordStatus.STARTED);
+        return parkingRecord.map(this::fromParkingRecord)
+            .<VehicleNotParkedException>orElseThrow(() -> {
```

Comment:
ten generic jest potrzebny?

Reply:
Normalnie powiedziałbym że oczywiście nie, ale przy próbie kompilacji dostałem `'Error:(59, 65) java: unreported exception java.lang.Throwable; must be caught or declared to be thrown'` mimo że użyłem Javy 1.8.0_162, wygląda to jak bug który już dawno był załatany [https://stackoverflow.com/questions/25523375/java8-lambdas-and-exceptions](https://stackoverflow.com/questions/25523375/java8-lambdas-and-exceptions) ale mimo to się pojawił